confgenerator/confgenerator_test.go

```diff
@@ -292,3 +272,27 @@ func TestMain(m *testing.M) {
 	}
 	os.Exit(m.Run())
 }
+
+func init() {
+	testResource := resourcedetector.GCEResource{
+		Project:       "test-project",
+		Zone:          "test-zone",
+		Network:       "test-network",
+		Subnetwork:    "test-subnetwork",
+		PublicIP:      "test-public-ip",
+		PrivateIP:     "test-private-ip",
+		InstanceID:    "test-instance-id",
+		InstanceName:  "test-instance-name",
+		Tags:          "test-tag",
+		MachineType:   "test-machine-type",
+		Metadata:      map[string]string{"test-key": "test-value"},
+		Label:         map[string]string{"test-label-key": "test-label-value"},
+		InterfaceIPv4: map[string]string{"test-interface": "test-interface-ipv4"},
+	}
+
+	// Set up the test environment with mocked data.
+	confgenerator.MetadataResource = testResource
+
+	// Enable experimental receivers.
+	os.Setenv("EXPERIMENTAL_RECEIVERS", "prometheus")
```

Comment:
If we're expanding this to be a more generic featuregate, we should probably not stop at receivers.

Maybe we rename it to `EXPERIMENTAL_FEATURES` and use gates like `prometheus_receiver` and `XYZ_processor` to toggle those components on and off.

Reply:
Done!